JavaScript/Topics/8.JSON/README.md

```diff
@@ -0,0 +1,216 @@
+
+## JSON Syntax
+
+JSON Syntax Rules:
+
+- Data is in name/value pairs
+- Data is separated by commas
+- Curly braces hold objects
+- Square brackets hold arrays
+
+### JSON Data
+
+JSON data is written as name/value pairs (aka key/value pairs). A name/value pair consists of a field name (in double quotes), followed by a colon, followed by a value.
+
+Example:
+
+```JSON
+"name":"Rahul"
+```
+
+### JSON - Evaluates to JavaScript Objects
+
+The JSON format is almost identical to JavaScript objects. In JSON, keys must be strings, written with double quotes.
+
+Example:
+
+#### JSON
+
+```JSON
+"name": "Rahul"
+```
+
+In JavaScript, keys can be strings, numbers, or identifier names.
+
+Example:
+
+#### Javascript
+
+```js
+name: "Rahul"
+```
+
+*Note: JSON names require double quotes.*
+
+### JSON Values
+
+In JSON, values must be one of the following data types:
+
+- string
+- number
+- object
+- array
+- boolean
+- null
+
+## JSON vs XML
+
+Both JSON and XML can be used to receive data from a web server.
+
+### JSON
+
+```JSON
+{
+     "students":
+          [
+               { "firstName":"Rahul", "lastName":"Roy" },
+               { "firstName":"Raj", "lastName":"Biswas" },
+               { "firstName":"Anjali", "lastName":"Mathur" }
+          ]
+}
+```
+
+### XML
+
+```xml
+<students>
+  <student>
+    <firstName>Rahul</firstName> <lastName>Roy</lastName>
+  </student>
+  <student>
+    <firstName>Raj</firstName> <lastName>Biswas</lastName>
+  </student>
+  <student>
+    <firstName>Anjali</firstName> <lastName>Mathur</lastName>
+  </student>
+</students>
+```
+
+### Similarities
+
+- JSON and XML are "self describing" (human readable)
+- JSON and XML are hierarchical (values within values)
+- JSON and XML can be parsed and used by lots of programming languages
+- JSON and XML can be fetched with an XMLHttpRequest
+
+### Differences
+
+- JSON doesn't use end tag
+- JSON is shorter
+- JSON is quicker to read and write
+- JSON can use arrays
+
+### Why JSON is Better Than XML?
+
+
+XML is much more difficult to parse than JSON. JSON is parsed into a ready-to-use JavaScript object.
+For AJAX applications, JSON is faster and easier than XML:
+
+Using XML:
+
+- Fetch an XML document
+- Use the XML DOM to loop through the document
+- Extract values and store in variables
+  
+Using JSON:
+
+- Fetch a JSON string
+- JSON.Parse the JSON string
+
+## JSON Data Types
+
+
+### Valid Data Types
+
+In JSON, values must be one of the following data types:
+- string
+- number
+- object (JSON object)
+- array
+- boolean
+- null
+
+JSON values cannot be one of the following data types:
+
+- a function
+- a date
+- undefined
+
+## JSON.parse()
+
+JSON is used to exchange data to/from a web server. When receiving data from a web server, the data is always a string. Parse the data with JSON.parse(), and the data becomes a JavaScript object.
+
+Example:
+
+Suppose we received this text from a web server:
```

Comment:
```suggestion
Suppose we received this JSON from a web server:
```

keep json everywhere whenever you are showing communication between webserver and client, because at some places you are saying it text and other places JSON, keep it json everywhere